packages/client/src/components/form/ReaderGenerator.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import React from 'react'
+import { QRReader } from '@opencrvs/components/src/IDReader/readers/QRReader/QRReader'
+import {
+  IFormData,
+  IFormField,
+  IFormFieldValue,
+  IFormSectionData,
+  Ii18nIDReaderFormField,
+  Ii18nRedirectFormField,
+  ReaderType
+} from '@client/forms'
+import { RedirectField } from './Redirect'
+import {
+  internationaliseFieldObject,
+  isReaderQR,
+  isReaderRedirect
+} from '@client/forms/utils'
+import {
+  messages,
+  tutorialMessages
+} from '@client/i18n/messages/views/qr-reader'
+import { useIntl } from 'react-intl'
+
+interface ReaderGeneratorProps {
+  readers: ReaderType[]
+  fields: IFormField[]
+  field: Ii18nIDReaderFormField
+  form: IFormSectionData
+  draft: IFormData
+  setFieldValue: (name: string, value: IFormFieldValue) => void
+}
+export const ReaderGenerator = ({
+  readers,
+  field,
+  form,
+  draft,
+  fields,
+  setFieldValue
+}: ReaderGeneratorProps) => {
+  const intl = useIntl()
+  return (
+    <>
+      {readers.map((reader) => {
+        const { type } = reader
+        if (isReaderQR(reader)) {
+          return (
+            <QRReader
+              key={type}
+              labels={{
+                button: intl.formatMessage(messages.button),
+                scannerDialogSupportingCopy: intl.formatMessage(
+                  messages.scannerDialogSupportingCopy
+                ),
+                tutorial: {
+                  cameraCleanliness: intl.formatMessage(
+                    tutorialMessages.cameraCleanliness
+                  ),
+                  distance: intl.formatMessage(tutorialMessages.distance),
+                  lightBalance: intl.formatMessage(
+                    tutorialMessages.lightBalance
+                  )
+                }
+              }}
+              onScan={(data) => {
+                setFieldValue(field.name, data)
+                setFieldValue('verified', 'pending')
+              }}
+              onError={(error) => console.error(error)}
+            />
+          )
+        } else if (isReaderRedirect(reader)) {
+          return (
+            <RedirectField
+              key={type}
+              form={form}
+              draft={draft}
+              fieldDefinition={
+                internationaliseFieldObject(
+                  intl,
+                  reader
+                ) as Ii18nRedirectFormField
+              }
+              fields={fields}
+              setFieldValue={setFieldValue}
+              isDisabled={false}
+            />
```

Comment:
Why is the redirect inside the reader and not just a generic form field?

Reply:
Thanks @naftis good point. Well, in the view component, e-signet button is inside the id-reader component
![image](https://github.com/user-attachments/assets/6ef617dd-7a4e-467a-9a6e-93f1cf439fb7)

Thought it would be better in code as well

Reply:
Ah makes sense. Should we consider renaming the field to a LINK_BUTTON or something? Got a bit confused as it isn't necessarily a redirect anymore

Reply:
Sounds good, thanks @naftis. I am on it